x-pack/filebeat/input/cometd/input.go

```diff
@@ -0,0 +1,199 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cometd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/elastic/beats/v7/filebeat/channel"
+	"github.com/elastic/beats/v7/filebeat/input"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+
+	bay "github.com/elastic/bayeux"
+)
+
+const (
+	inputName = "cometd"
+)
+
+// Run starts the input worker then returns. Only the first invocation
+// will ever start the worker.
+func (in *cometdInput) Run() {
+	var err error
+	in.workerOnce.Do(func() {
+		in.workerWg.Add(1)
+		go func() {
+			in.log.Info("Input worker has started.")
+			defer in.log.Info("Input worker has stopped.")
+			defer in.workerWg.Done()
+			defer in.workerCancel()
+			in.b = bay.Bayeux{}
+			in.creds, err = bay.GetSalesforceCredentials(in.authParams)
+			if err != nil {
+				in.log.Error("not able to get access token: ", err)
```

Comment:
Consider using the structured logger. For example:

```suggestion
				in.log.Errorw("not able to get access token", "error", err)
```

The our [ECS logging encoder](https://github.com/elastic/ecs-logging-go-zap) will output this in proper ECS format like `{"message": "not able to get access token", "error": {"message": err.Error()}}`.

Reply:
Oh, I will update accordingly. Thanks 👍